### PR TITLE
Build gitian using Ubuntu 18.04 based docker containers

### PR DESCRIPTION
**Pull request by sickpig**

### CODE REVIEW

**Comment by sickpig** [comment]
@gandrewstone taking out of WIP, PR is ready to go in imho.

File: doc/gitian-building-bu.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+git clone https://github.com/BitcoinUnlimited/BitcoinUnlimited.git
+git clone https://github.com/devrandom/gitian-builder.git
+cd gitian-builder
+bin/make-base-vm --suite bionic --arch amd64 --docker
+```
+
+Build the binaries
+------------------
+
+cd ~/src/gitian-builder
```
Comment (gandrewstone) [inline doc/gitian-building-bu.md:29]: needs ```bash

Comment (sickpig) [reply doc/gitian-building-bu.md:29]: right, fixing.

File: doc/gitian-building-bu.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+Prerequisite
+-------------
+
+These are steps that as to be executed once and that don't need to be repeated for every new gitian build process.
+
+```bash
+sudo apt install git apt-cacher-ng ruby docker.io
+sudo usermod -a -G docker $USER
+cd ~/src
```
Comment (gandrewstone) [inline doc/gitian-building-bu.md:16]: ```gnome-session-quit```  # log out and back in so group change takes effect

Comment (sickpig) [reply doc/gitian-building-bu.md:16]: Actually issuing this command:

```bash
exec su -l $USER
```
from the same terminal that will be used to run the `bin/gbuild` should do trick.

File: contrib/gitian-descriptors/gitian-win.yml [hunk 3]
```diff
@@ -2,13 +2,13 @@
 name: "BUcash-win-1.5"
 enable_cache: true
 suites:
-- "trusty"
```
Comment (gandrewstone) [inline contrib/gitian-descriptors/gitian-win.yml:5]: bionic

File: depends/packages/bdb.mk [hunk 4]
```diff
@@ -9,7 +9,7 @@ define $(package)_set_vars
 $(package)_config_opts=--disable-shared --enable-cxx --disable-replication
 $(package)_config_opts_mingw32=--enable-mingw
 $(package)_config_opts_linux=--with-pic
-$(package)_cxxflags=-std=c++11
+$(package)_cxxflags=-std=c++1y
```
Comment (gandrewstone) [inline depends/packages/bdb.mk]: c++14, 1y is the pre-standard flag

File: depends/packages/boost.mk [hunk 5]
```diff
@@ -20,7 +20,7 @@ $(package)_archiver_$(host_os)=$($(package)_ar)
 $(package)_toolset_darwin=darwin
 $(package)_archiver_darwin=$($(package)_libtool)
 $(package)_config_libraries=chrono,filesystem,program_options,system,thread,test
-$(package)_cxxflags=-std=c++11 -fvisibility=hidden
+$(package)_cxxflags=-std=c++1y -fvisibility=hidden
```
Comment (gandrewstone) [inline depends/packages/boost.mk]: c++14

File: depends/packages/protobuf.mk [hunk 6]
```diff
@@ -4,7 +4,7 @@ $(package)_download_path=$(native_$(package)_download_path)
 $(package)_file_name=$(native_$(package)_file_name)
 $(package)_sha256_hash=$(native_$(package)_sha256_hash)
 $(package)_dependencies=native_$(package)
-$(package)_cxxflags=-std=c++11
+$(package)_cxxflags=-std=c++1y
```
Comment (gandrewstone) [inline depends/packages/protobuf.mk]: 14

File: depends/packages/bdb.mk (hunk 4 above)
Comment (sickpig) [reply depends/packages/bdb.mk]: you are right indeed.

File: depends/packages/boost.mk (hunk 5 above)
Comment (sickpig) [reply depends/packages/boost.mk]: ditto

File: depends/packages/protobuf.mk (hunk 6 above)
Comment (sickpig) [reply depends/packages/protobuf.mk]: ditto

File: contrib/gitian-descriptors/gitian-win.yml (hunk 3 above)
Comment (sickpig) [reply contrib/gitian-descriptors/gitian-win.yml:5]: fixed

File: doc/gitian-building-bu.md (hunk 1 above)
Comment (sickpig) [reply doc/gitian-building-bu.md:29]: fixed

File: doc/gitian-building-bu.md (hunk 2 above)
Comment (sickpig) [reply doc/gitian-building-bu.md:16]: fixed

File: depends/packages/bdb.mk (hunk 4 above)
Comment (sickpig) [reply depends/packages/bdb.mk]: fixed

File: depends/packages/protobuf.mk (hunk 6 above)
Comment (sickpig) [reply depends/packages/protobuf.mk]: fixed

File: depends/packages/boost.mk (hunk 5 above)
Comment (sickpig) [reply depends/packages/boost.mk]: fixed

**Pull request merged**